src/scanspec/core.py

```diff
@@ -482,6 +507,68 @@ def _merge_frames(
     )
 
 
+def stack2dimension(
+    dimensions: list[Dimension[Axis]],
+    indices: npt.NDArray[np.signedinteger] | None = None,
+    lengths: npt.NDArray[np.signedinteger] | None = None,
+) -> Dimension[Axis]:
+    if lengths is not None:
+        lengths = lengths
+    else:
+        lengths = np.array([len(f) for f in dimensions])
```

Comment:
Nit: in general I prefer
```suggestion
    if lengths is None:
        lengths = np.array([len(f) for f in dimensions])
```

Reply:
Done!